libs/tags/azure.go

```diff
@@ -0,0 +1,25 @@
+package tags
+
+import (
+	"regexp"
+
+	"golang.org/x/text/unicode/rangetable"
+)
+
+// All characters that may not be used in Azure tag keys.
+var azureForbiddenChars = rangetable.New('<', '>', '*', '&', '%', ';', '\\', '/', '+', '?')
+
+var azureTag = &tag{
+	keyLength:  512,
+	keyPattern: regexp.MustCompile(`^[^<>\*&%;\\\/\+\?]*$`),
```

Comment:
don't we need to escape `^` inside the [] brackets?

Reply:
The `^` inside a character class means to negate it; i.e. characters other than the following ones.

Also see https://pkg.go.dev/regexp/syntax.

Reply:
ah, got it. I confused myself and thought that `^` is in the forbidden chars and the first `^` is for negation, not start of the string